source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -214,14 +222,28 @@ HttpConnectionManagerFilterConfigFactory::createFilterFactoryFromProtoTyped(
   // reference count.
   // Keep in mind the lambda capture list **doesn't** determine the destruction order, but it's fine
   // as these captured objects are also global singletons.
-  return [singletons, filter_config, &context](Network::FilterManager& filter_manager) -> void {
-    filter_manager.addReadFilter(Network::ReadFilterSharedPtr{new Http::ConnectionManagerImpl(
+  return [singletons, filter_config, &context,
+          clear_hop_by_hop_headers](Network::FilterManager& filter_manager) -> void {
+    auto hcm = new Http::ConnectionManagerImpl(
```

Comment:
make_shared?

Reply:
We could make the shared readfiltersharedptr but then would need to cast it back for the call to setClearHopByHopResponseHeaders (which isn't in the interface).  I chose raw pointer and making shared at addReadFilter but lmk if you prefer casts.

Reply:
Can't you just make a `shared_ptr<Http::ConnectionManagerImpl>` and then implicit casting will handle the upcast?

Reply:
good call, on it!